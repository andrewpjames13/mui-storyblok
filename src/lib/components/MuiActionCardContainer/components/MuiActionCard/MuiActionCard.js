import React from 'react';
import PropTypes from 'prop-types';
import MuiTypography from '../../../MuiTypography/MuiTypography';
import MuiIcon from '../../../MuiIcon/MuiIcon';
import styles from './MuiActionCard.module.scss';
import StoryBlok from '../../../../utils/Storyblok';

const MuiActionCard = ({
  header,
  icon,
  text,
  height,
  width,
  rootClass,
}) => {
  const classes = StoryBlok.anchorOrginToObj(rootClass);

  return (
    <div style={{ height, width, ...classes }} className={styles.actionCard}>
      <div className={styles.actionIcon}>
        <MuiIcon {...icon[0]} />
      </div>
      <div className={styles.actionHeader}>
        <MuiTypography {...header[0]} />
      </div>
      <div className={styles.actionBody}>
        <MuiTypography {...text[0]} />
      </div>
    </div>
  );
};

export default MuiActionCard;

MuiActionCard.propTypes = {
  /**
   * storyblok multiselect of css classes
  * */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * Cards passed to MuiActionCard to render
   * Component: MuiTypography
   */
  header: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
  /**
   * Cards passed to MuiActionCard to render
   * Component: MuiIcon
   */
  icon: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
  /**
   * Cards passed to MuiActionCard to render
   * Component: MuiTypography
   */
  text: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
  /**
   * Passed down from ActionCardContainer
   * Height of the each Action Card.
   */
  height: PropTypes.string,
  /**
   * Passed down from ActionCardContainer
   * Width of the each Action Card.
   */
  width: PropTypes.string,
};

MuiActionCard.defaultProps = {
  height: '100px',
  width: '200px',
  rootClass: [],
};
