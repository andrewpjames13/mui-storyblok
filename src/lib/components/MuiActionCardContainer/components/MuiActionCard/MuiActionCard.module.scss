.actionCard {
  padding: 10px;
  display: grid;
  height: 100%;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 10px;
  grid-template-areas:
    "icon header header header"
    "icon text text text";
  transition: 0.5s;

  &:hover {
    background-color: #C7EBFA;
    cursor: pointer;
  }
}

.actionIcon {
  grid-area: icon;
}

.actionHeader {
  grid-area: header;
}

.actionBody {
  grid-area: text;
}
