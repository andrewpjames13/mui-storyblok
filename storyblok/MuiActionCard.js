module.exports = {
  "name": "MuiActionCard",
  "display_name": null,
  "updated_at": "2020-06-03T15:51:04.447Z",
  "schema": {
    "header": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "MuiTypography"
      ]
    },
    "icon": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "MuiIcon"
      ]
    },
    "text": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "MuiTypography"
      ]
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiActionCard",
  "component_group_uuid": null
}