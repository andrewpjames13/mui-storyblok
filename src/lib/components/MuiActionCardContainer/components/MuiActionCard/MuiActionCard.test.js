import React from 'react';
import { shallow } from 'enzyme';
import MuiActionCard from './MuiActionCard';

function setup() {
  const props = {
    height: '100px',
    width: '100px',
    header: [{
      component: 'MuiTypography',
      content: [{
        component: 'MuiText',
        text: 'Header of a card',
      }],
    }],
    icon: [{
      component: 'MuiIcon',
      iconName: 'star',
    }],
    text: [{
      component: 'MuiTypography',
      content: [{
        component: 'MuiText',
        text: 'this is the text of an action card body.',
      }],
    }],
  };
  const comp = shallow(<MuiActionCard {...props} />);
  return { comp, props };
}

describe('<MuiActionCard />', () => {
  it('renders MuiActionCard', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });
});
