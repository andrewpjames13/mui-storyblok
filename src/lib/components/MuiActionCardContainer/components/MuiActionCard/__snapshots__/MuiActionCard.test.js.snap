// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiActionCard /> snapshot 1`] = `
<div
  className="actionCard"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div
    className="actionIcon"
  >
    <span
      aria-hidden={true}
      className="material-icons MuiIcon-root makeStyles-root-1 MuiIcon-colorSecondary"
    >
      star
    </span>
  </div>
  <div
    className="actionHeader"
  >
    <div
      className="MuiTypography-root makeStyles-root-2 MuiTypography-body1"
    >
      <span
        className="MuiTypography-root makeStyles-root-3 MuiTypography-body1"
      >
         Header of a card 
      </span>
    </div>
  </div>
  <div
    className="actionBody"
  >
    <div
      className="MuiTypography-root makeStyles-root-4 MuiTypography-body1"
    >
      <span
        className="MuiTypography-root makeStyles-root-5 MuiTypography-body1"
      >
         undefined 
      </span>
    </div>
  </div>
</div>
`;
